File: examples/c/sum_of_digits.c
#include <assert.h>
#include <stdio.h>

static int sum_of_digits(unsigned int number) {
    int sum = 0;

    do {
        sum += (int)(number % 10U);
        number /= 10U;
    } while (number > 0U);

    return sum;
}

int main(void) {
    assert(sum_of_digits(507U) == 12);
    assert(sum_of_digits(0U) == 0);
    printf("sum_of_digits: ok\n");
    return 0;
}

File: examples/java/SumOfDigits.java
public class SumOfDigits {
    static int sumOfDigits(int number) {
        if (number < 0) {
            throw new IllegalArgumentException("number must be non-negative");
        }

        int sum = 0;
        do {
            sum += number % 10;
            number /= 10;
        } while (number > 0);

        return sum;
    }

    static void check(boolean condition, String message) {
        if (!condition) {
            throw new AssertionError(message);
        }
    }

    public static void main(String[] args) {
        check(sumOfDigits(507) == 12, "507 should produce 12");
        check(sumOfDigits(0) == 0, "0 should produce 0");
        System.out.println("sumOfDigits: ok");
    }
}
